README.rst
Python Client for Assured Workloads for Government API
======================================================

|stable| |pypi| |versions|

`Assured Workloads for Government API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-assured-workloads.svg
   :target: https://pypi.org/project/google-cloud-assured-workloads/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-assured-workloads.svg
   :target: https://pypi.org/project/google-cloud-assured-workloads/
.. _Assured Workloads for Government API: https://cloud.google.com/assured-workloads/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/assuredworkloads/latest
.. _Product Documentation:  https://cloud.google.com/assured-workloads/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Assured Workloads for Government API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Assured Workloads for Government API.:  https://cloud.google.com/assured-workloads/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-assured-workloads


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-assured-workloads

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Assured Workloads for Government API
   to see other available methods on the client.
-  Read the `Assured Workloads for Government API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Assured Workloads for Government API Product documentation:  https://cloud.google.com/assured-workloads/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [1.1.0](https://www.github.com/googleapis/python-assured-workloads/compare/v1.0.0...v1.1.0) (2022-01-07)


### Features

* EU Regions and Support With Sovereign Controls ([#110](https://www.github.com/googleapis/python-assured-workloads/issues/110)) ([a0676ef](https://www.github.com/googleapis/python-assured-workloads/commit/a0676ef0f3d35d27886b7e624973de7942b34214))

## [1.0.0](https://www.github.com/googleapis/python-assured-workloads/compare/v0.8.0...v1.0.0) (2021-11-01)


### Features

* bump release level to production/stable ([#92](https://www.github.com/googleapis/python-assured-workloads/issues/92)) ([c346fbb](https://www.github.com/googleapis/python-assured-workloads/commit/c346fbb3c3c4388100ba08d5a30889e96727e351))


### Bug Fixes

* **deps:** drop packaging dependency ([2bc0174](https://www.github.com/googleapis/python-assured-workloads/commit/2bc01744f9bbc48f3e5d1de1dd196571fc1494db))
* **deps:** require google-api-core >= 1.28.0 ([2bc0174](https://www.github.com/googleapis/python-assured-workloads/commit/2bc01744f9bbc48f3e5d1de1dd196571fc1494db))


### Documentation

* list oneofs in docstring ([2bc0174](https://www.github.com/googleapis/python-assured-workloads/commit/2bc01744f9bbc48f3e5d1de1dd196571fc1494db))

## [0.8.0](https://www.github.com/googleapis/python-assured-workloads/compare/v0.7.2...v0.8.0) (2021-10-11)


### Features

* add context manager support in client ([#94](https://www.github.com/googleapis/python-assured-workloads/issues/94)) ([1dd0e98](https://www.github.com/googleapis/python-assured-workloads/commit/1dd0e982ae7268996ecbf534ed310e446f2d0070))
* add trove classifier for python 3.10 ([#97](https://www.github.com/googleapis/python-assured-workloads/issues/97)) ([4db7fe0](https://www.github.com/googleapis/python-assured-workloads/commit/4db7fe0f9655b09c47eef5142a8c0a7ba0e270d9))

### [0.7.2](https://www.github.com/googleapis/python-assured-workloads/compare/v0.7.1...v0.7.2) (2021-09-30)


### Bug Fixes

* improper types in pagers generation ([a78556b](https://www.github.com/googleapis/python-assured-workloads/commit/a78556b3481b7e000a209ce0495b52495769cb9a))

### [0.7.1](https://www.github.com/googleapis/python-assured-workloads/compare/v0.7.0...v0.7.1) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([5c4dd09](https://www.github.com/googleapis/python-assured-workloads/commit/5c4dd0988b444175a3e6cc5c4b3e183d56cefaba))

## [0.7.0](https://www.github.com/googleapis/python-assured-workloads/compare/v0.6.0...v0.7.0) (2021-09-20)


### Features

* assuredworkloads v1 public protos ([#82](https://www.github.com/googleapis/python-assured-workloads/issues/82)) ([04dd627](https://www.github.com/googleapis/python-assured-workloads/commit/04dd627d5d3862b055d661c2a1bf9a0f6b5fc4e4))

## [0.6.0](https://www.github.com/googleapis/python-assured-workloads/compare/v0.5.0...v0.6.0) (2021-08-30)


### Features

* Add Canada regions and support compliance regime ([#73](https://www.github.com/googleapis/python-assured-workloads/issues/73)) ([b62f7c7](https://www.github.com/googleapis/python-assured-workloads/commit/b62f7c720d198741673cc93d452d0ac9067cd3c3))
* display_name is added to ResourceSettings ([b62f7c7](https://www.github.com/googleapis/python-assured-workloads/commit/b62f7c720d198741673cc93d452d0ac9067cd3c3))
* resource_settings is added to CreateWorkloadOperationMetadata ([b62f7c7](https://www.github.com/googleapis/python-assured-workloads/commit/b62f7c720d198741673cc93d452d0ac9067cd3c3))
* ResourceType CONSUMER_FOLDER and KEYRING are added ([b62f7c7](https://www.github.com/googleapis/python-assured-workloads/commit/b62f7c720d198741673cc93d452d0ac9067cd3c3))


### Bug Fixes

* billing_account is now optional in Workload ([b62f7c7](https://www.github.com/googleapis/python-assured-workloads/commit/b62f7c720d198741673cc93d452d0ac9067cd3c3))
* ResourceType CONSUMER_PROJECT is deprecated ([b62f7c7](https://www.github.com/googleapis/python-assured-workloads/commit/b62f7c720d198741673cc93d452d0ac9067cd3c3))

## [0.5.0](https://www.github.com/googleapis/python-assured-workloads/compare/v0.4.2...v0.5.0) (2021-07-28)


### Features

* Add EU Regions And Support compliance regime ([#67](https://www.github.com/googleapis/python-assured-workloads/issues/67)) ([a370ad5](https://www.github.com/googleapis/python-assured-workloads/commit/a370ad5c1c7525544f3e5a83e84e0c05ed1851e2))

### [0.4.2](https://www.github.com/googleapis/python-assured-workloads/compare/v0.4.1...v0.4.2) (2021-07-27)


### Bug Fixes

* enable self signed jwt for grpc ([#64](https://www.github.com/googleapis/python-assured-workloads/issues/64)) ([c7e4331](https://www.github.com/googleapis/python-assured-workloads/commit/c7e43317be9e68508449a0f9cb548d1bd5904f1e))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#60](https://www.github.com/googleapis/python-assured-workloads/issues/60)) ([b161d65](https://www.github.com/googleapis/python-assured-workloads/commit/b161d658c8cdf294f72181b368e9e8df3529c392))


### Miscellaneous Chores

* release as 0.4.2 ([#65](https://www.github.com/googleapis/python-assured-workloads/issues/65)) ([8f8f538](https://www.github.com/googleapis/python-assured-workloads/commit/8f8f53852fd2e3ae4a917cdd7c37125fb01043a4))

### [0.4.1](https://www.github.com/googleapis/python-assured-workloads/compare/v0.4.0...v0.4.1) (2021-07-20)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#59](https://www.github.com/googleapis/python-assured-workloads/issues/59)) ([5113968](https://www.github.com/googleapis/python-assured-workloads/commit/5113968fa3e779a1e1d69f3642d9cd2f7ebcbe91))

## [0.4.0](https://www.github.com/googleapis/python-assured-workloads/compare/v0.3.1...v0.4.0) (2021-06-30)


### Features


owl-bot-staging/v1beta1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9', '3.10'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/assuredworkloads_v1beta1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.9')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.9')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
